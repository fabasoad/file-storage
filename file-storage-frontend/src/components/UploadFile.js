import React from 'react';

export default class UploadFile extends React.Component {
    state = {
        error: undefined
    };
    handleUploadFile = (e) => {
        e.preventDefault();
        const file = e.target.elements.file.value.trim();
        const error = this.props.handleUploadFile(file);

        this.setState(() => ({ error }));

        if (!error) {
            e.target.elements.file.value = '';
        }   
    };
    render() {
        return (
            <div>
                {this.state.error && <p className="upload-file-error">{this.state.error}</p>}
                <form className="upload-file" onSubmit={this.handleUploadFile}>
                    <input className="upload-file__input" type="file" name="file" />
                    <button type="submit" className="button">Upload</button>
                </form>
            </div>
        );
    }
}