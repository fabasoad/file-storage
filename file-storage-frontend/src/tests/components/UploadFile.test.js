import React from 'react';
import { shallow } from 'enzyme';
import UploadFile from '../../components/UploadFile';

let wrapper, handleUploadFile;

beforeEach(() => {
    handleUploadFile = jest.fn();
    wrapper = shallow(<UploadFile handleUploadFile={handleUploadFile} />);
});

test('should render UploadFile correctly', () => {
    expect(wrapper).toMatchSnapshot();
});

test('should handle handleUploadFile', () => {
    const expectedFileName = 'file1.txt';
    wrapper.find('form').prop('onSubmit')({
        preventDefault: () => {},
        target: { elements: { file: { value: expectedFileName } } }
    });
    expect(handleUploadFile).toHaveBeenLastCalledWith(expectedFileName);
});