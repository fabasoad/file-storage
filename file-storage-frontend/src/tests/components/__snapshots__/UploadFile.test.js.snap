// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render UploadFile correctly 1`] = `
<div>
  <form
    className="upload-file"
    onSubmit={[Function]}
  >
    <input
      className="upload-file__input"
      name="file"
      type="file"
    />
    <button
      className="button"
      type="submit"
    >
      Upload
    </button>
  </form>
</div>
`;
